lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -316,16 +254,93 @@ class FontSize extends Gatherer {
     passContext.driver.on('CSS.styleSheetAdded', onStylesheetAdd);
 
     await Promise.all([
+      passContext.driver.sendCommand('DOMSnapshot.enable'),
       passContext.driver.sendCommand('DOM.enable'),
       passContext.driver.sendCommand('CSS.enable'),
     ]);
 
-    const {
-      totalTextLength,
-      visitedTextLength,
-      failingTextLength,
-      failingNodes,
-    } = await FontSize.fetchNodesToAnalyze(passContext.driver);
+    // We need to find all TextNodes that do not have legible text. DOMSnapshot.captureSnapshot is the
+    // fastest way to get the computed styles of every Node. Bonus, it allows for whitelisting properties.
+    // Once a bad TextNode is identified, its parent Node is needed. DOMSnapshot.captureSnapshot doesn't
+    // give the entire Node object, so DOM.getFlattenedDocument is used. The only connection between a snapshot
+    // Node and an actual Protocol Node is backendId, so that is used to join the two data structures.
+    const snapshot = await passContext.driver.sendCommand('DOMSnapshot.captureSnapshot', {
+      computedStyles: ['font-size'],
+    });
+
+    // Makes the strings access code easier to read.
+    /** @param {number} index */
+    const lookup = (index) => snapshot.strings[index];
+
+    // Locate the document under analysis.
+    // TODO: this needs to use frameId
+    const doc = snapshot.documents.find(doc => lookup(doc.documentURL) === passContext.url);
```

Comment:
I need help here.

Reply:
Hm, is it not true that "The DOMNode at index 0 corresponds to the root document"? Should poke devtools folks to update that then :)

AFAIK, the only main frame id we have/use is in the trace (the mainFrameIds refactor you just did), and I don't think we're collecting it anywhere else that'd be readily available to a gatherer.

You could query it with `Page.getResourceTree` `resourceTreeResponse.frameTree.frame.id` or use the network records to get at the frame id if we need to